detox/src/devices/runtime/drivers/android/AndroidDriver.js

```diff
@@ -233,20 +233,45 @@ class AndroidDriver extends DeviceDriverBase {
     await this.invocationManager.execute(call);
   }
 
-  _getInstallPaths(_binaryPath, _testBinaryPath) {
-    const binaryPath = getAbsoluteBinaryPath(_binaryPath);
-    const testBinaryPath = _testBinaryPath ? getAbsoluteBinaryPath(_testBinaryPath) : this._getTestApkPath(binaryPath);
+  _getAppInstallPaths(_appBinaryPath, _testBinaryPath) {
+    const appBinaryPath = getAbsoluteBinaryPath(_appBinaryPath);
+    const testBinaryPath = _testBinaryPath ? getAbsoluteBinaryPath(_testBinaryPath) : this._getTestApkPath(appBinaryPath);
     return {
-      binaryPath,
+      appBinaryPath,
       testBinaryPath,
     };
   }
 
+  async _validateAppBinaries(appBinaryPath, testBinaryPath) {
+    if (await this.aapt.isTestAPK(appBinaryPath)) {
+      logger.warn(new DetoxRuntimeError({
+        message: `App APK at path ${appBinaryPath} was detected as the *test* APK!`,
+        hint: 'Your binary-path was probably wrongly set in the active Detox configuration',
+      }).message);
+    }
+
+    if (!await this.aapt.isTestAPK(testBinaryPath)) {
+      logger.warn(new DetoxRuntimeError({
+        message: `Test APK at path ${testBinaryPath} was detected as the *app* APK!`,
+        hint: 'Your test binary-path was probably wrongly set in the active Detox configuration',
+      }).message);
+    }
+  }
+
+  async _installAppBinaries(appBinaryPath, testBinaryPath) {
+    await this.adb.install(this.adbName, appBinaryPath);
+    await this.adb.install(this.adbName, testBinaryPath);
+  }
+
   _getTestApkPath(originalApkPath) {
-    const testApkPath = APKPath.getTestApkPath(originalApkPath);
+    const testApkPath = apkUtils.getTestApkPath(originalApkPath);
 
     if (!fs.existsSync(testApkPath)) {
-      throw new DetoxRuntimeError(`'${testApkPath}' could not be found, did you run './gradlew assembleAndroidTest'?`);
+      throw new DetoxRuntimeError({
+        message: `The test APK could not be found at path: '${testApkPath}'`,
+        hint: 'Try running the detox build command, and make sure it was configured to execute a build command (e.g. \'./gradlew assembleAndroidTest\')\n' +
+          'For further assistance, visit the Android setup guide: https://github.com/wix/Detox/blob/master/docs/Introduction.Android.md',
```

Comment:
It may be better to refer to the website doc page instead of the GH one:
https://wix.github.io/Detox/docs/introduction/android

Reply:
I agree but I didn't want to שעטנז it :-)
we should probably change things across the board, if we ever.

Reply:
I tend to agree with @asafkorem. Amit, you're adding here a *new* link to the error description, so he's right.

Reply:
As a Detox user, would feel odd if Detox were to refer me to the github docs and the website intermittently. Nevertheless, my opinion here is not super strong, so I will update this if you insist on it :-)